### PR TITLE
Update for Allsorts 0.4

### PR DESCRIPTION
**Pull request by wezm**
* Update for Allsorts 0.4.
* Add `--glyph-names` and `--encodings` options to dump command.
* Add option to print head table.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,12 @@ name = "allsorts"
...
-allsorts = "0.3.0"
 atty = "0.2.13"
 bitreader = "0.3.2"
 encoding_rs = "0.8.16"
 gumdrop = "0.7.0"
 itertools = "0.8.0"
 png = "0.15.3"
+
+[dependencies.allsorts]
+version = "0.4.0"
```
Comment (brendanzab) [inline Cargo.toml:21]: Why this change? Just curious :)

**Review by brendanzab — APPROVED** [review]
Looks good!

File: Cargo.toml (hunk 1 above)
Comment (wezm) [reply Cargo.toml:21]: > Why this change?

Minor convenience. I often use `allsorts-tools` to exercise new `allsorts` functionality.

The transformation from:

```toml
[dependencies.allsorts]
version = "0.4.0"
```

to 

```toml
[dependencies.allsorts]
version = "0.4.0"
path = "../path/to/allsorts"
```

Is simpler than:

```toml
allsorts = "0.3.0"
```

to 

```toml
allsorts = { version = "0.3.0", path = "../path/to/allsorts" }
